src/compiler/crystal/types.cr

```diff
@@ -38,6 +38,19 @@ module Crystal
       program
     end
 
+    # The namespaces this type belongs to, excluding the `Program` itself.
+    def namespaces : Array(ModuleType)
+      namespace_types = [] of ModuleType
```

Comment:
Can't we enumerate the namespaces instead of creating an array for each type?

Reply:
Possibly?  How were you envisioning that working?

Reply:
An `each_namespace` method that yields. Then you can't use `any? ` and it's a bit more code to use, but it's fine.